cmd/juju/application/upgradecharm.go

```diff
@@ -492,10 +501,12 @@ func (c *upgradeCharmCommand) addCharm(
 	config *config.Config,
 	oldURL *charm.URL,
 	charmRef string,
+	deployedSeries string,
 ) (charmstore.CharmID, *macaroon.Macaroon, error) {
 	var id charmstore.CharmID
-	// Charm may have been supplied via a path reference.
-	ch, newURL, err := charmrepo.NewCharmAtPathForceSeries(charmRef, oldURL.Series, c.ForceSeries)
+	// Charm may have been supplied via a path reference. If so, build a
+	// local charm URL from the deployed series.
+	ch, newURL, err := charmrepo.NewCharmAtPathForceSeries(charmRef, deployedSeries, c.ForceSeries)
```

Comment:
Done. I was fairly convinced that changing to the `deployedSeries` would work for generating the `newURL` wasn't too sure about the additional 3 references below that. I've looked at it and indeed I feel you are correct that they too can be safely replaced by the `deployedSeries`. Thank you @howbazaar.